### DISCUSSION TITLE
Modulo 2 Kata 2

### DISCUSSION THREAD

**Discussion by AndresAVargas** [LaunchX-InnovaccionVirtual/CursoIntroPython#22] [Q&A] [answered]
Compañero estoy estudiando el modulo 2 para realizar la kata 2 pero no paso de aqui, no se nada de python
![image](https://user-images.githubusercontent.com/88987587/153075865-aeb98f58-96c6-4b83-8b00-e5a14b049de6.png)
se que ya respondieron esta duda muchas veces pero ya no me aparece la discusiones de hace rato

**Comment by FernandaOchoa** [comment, marked as answer]
Hola, acabo de actualizar el repositorio para que revisen.

Aquí veo varios detalles: 

* Está mal escrito el primer comando por lo que aunque continúes con lo demás no va a funcionar ya que trabaja línea por línea. En la documentación está el comando escrito correctamente, puedes copiarlo y pegarlo dándo **Clic derecho** en tu mouse.

De igual manera aquí lo dejo: 
`python -m venv env`

* El siguiente paso es ir al entorno virtual, por lo que también puedes copiar el comando desde el documento que está para auxiliar la kata.
* Por último, sabrás que está correcto, cuándo se logra visualizar al principio la palabra (env)

Suerte! 💖 


![image](https://user-images.githubusercontent.com/9124597/153077221-8420d46c-2ca2-42ff-9061-db41cd212202.png)

**Reply by alb333z** [reply]
Hola fer como estas? Mi nombre es alberto tengo el siguiente problema. Ya se crea el directorio env solamente al querer activar el entorno no e podido. Ya lo intente de diferentes maneras espero me puedas ayudar por favor. 

![duda_curso](https://user-images.githubusercontent.com/48806691/153090326-9cdeeabe-0408-4e49-99f2-be0a9934e83a.jpg)

**Reply by alb333z** [reply]
![image](https://user-images.githubusercontent.com/48806691/153093039-12621903-bbdc-4699-8f3b-19a702a2685c.png)
Listo :) era la consola!

**Reply by lethal-dev** [reply]
Hola Alberto creo que te puedo ayudar
1. Primero debes de estar dentro de la carpeta de Scripts ya sea a través de la consola o de la ventana:
![1](https://user-images.githubusercontent.com/70924817/153092807-9a4e839a-6a79-468f-9444-60252d9dea57.png)

2. Después si entraste por la ventana das click derecho y ejecutas la terminal:
![2](https://user-images.githubusercontent.com/70924817/153092899-37d17f92-8625-494e-b2d2-1a92b6593167.png)

3. Una vez estando ahí ejecutas activate esta captura es de la terminal de Windows si utilizas Bash creo que es /.activate:
![3](https://user-images.githubusercontent.com/70924817/153093077-9649d85c-f609-4802-93a6-717e345ec2c1.png)

4. Y por ultimo se debe de ver así (dice env al inicio):
![4](https://user-images.githubusercontent.com/70924817/153093134-532fbb81-69b9-46cf-ad8e-276f9250947b.png)

**Reply by enrique-mendoza** [reply]
Hola @alb333z.

Por lo que observo en tu screenshot estas utilizando la consola bash de Git para ejecutar comandos. Prueba utilizando la consola CMD de Windows y ejecutarla como administrador.

Quedo atento a cualquier comentario o corrección. :D

**Reply by alb333z** [reply]
Muchas gracias! @RafaCEE , @enrique-mendoza

**Reply by maxmiau** [reply]
Buen dia a todos, para este kata que es lo que se va a subir a github como comprobante de haber hecho la actividad?

**Reply by Angel-Guerra** [reply]
> ![image](https://user-images.githubusercontent.com/48806691/153093039-12621903-bbdc-4699-8f3b-19a702a2685c.png) Listo :) era la consola!

si lo mismo me paso

**Reply by JonathanGrimaldo** [reply]
Si estan en bash en windows, me funciono lo siguiente, pueden probar

`source env/Scripts/activate`

![image](https://user-images.githubusercontent.com/50802445/153691772-ad1ca694-5876-41f3-ada5-408d8b3e0f35.png)

**Comment by DiegoGomezL03** [comment]
Buenas tardes @FernandaOchoa, yo tengo el siguiente problema. 

Ya me dirigí a configuración  y revise los alias de las aplicaciones y todo correcto, pero aun sigo sin poder ejecutar el entorno virtual. ¿Sabrás que estoy haciendo mal?

Muchas gracias por tu ayuda.

![env](https://user-images.githubusercontent.com/87287131/153081696-68a5e01c-2783-4449-a5fc-93fd95e0f410.PNG)

**Reply by jdcarrillo123** [reply]
Intenta con:
python3 -m venv env

**Reply by DiegoGomezL03** [reply]
igual intente, y sale el mismo error.

**Reply by jdcarrillo123** [reply]
Asegúrese de tener instalado el python
```python --version```

**Reply by rickygzz** [reply]
Hola Diego,

Imagino que descargaste la última versión de Python 3.10.2 (https://www.python.org/downloads/). Al instalarla te dio a elegir el folder de instalación

![image](https://user-images.githubusercontent.com/36200981/153082860-8d9c4a53-02a3-4324-b545-ee4f395d97d3.png)

En mi caso me sugiere la ruta `C:\Users\PC\AppData\Local\Programs\Python\Python310\`   (mi nombre de usuario es PC, si tu nombre de usuario para entrar a Windows es "Diego" entonces la ruta sugerida sería `C:\Users\Diego\AppData\Local\Programs\Python\Python310\`). Cualquiera que haya sido la ruta de instalación, comprueba en el explorador de Windows si esa ruta contiene los archivos ejecutables.

![image](https://user-images.githubusercontent.com/36200981/153083356-d813b9cf-1eaf-4a11-a539-3eff0ed9c71c.png)

Una vez confirmada la ruta, copiala con Ctrl + C. En el explorador de Windows, ubica `Mi PC`, da clic derecho, Propiedades, Configuraciones de sistema avanzadas

![image](https://user-images.githubusercontent.com/36200981/153083837-cf370ce3-7357-48a7-a254-0d80196f51ee.png)

Te abre un cuadro de diálogo, en el tab Avanzado, da clic en el botón Variables de Entorno

![image](https://user-images.githubusercontent.com/36200981/153084017-d8fe9d69-6302-4e01-b391-cef62278b18c.png)

En el nuevo cuadro de diálogo, selecciona la variable Path y editar

![image](https://user-images.githubusercontent.com/36200981/153084281-868f8d8c-ae2d-4a7b-bac8-6725bf1adce5.png)

Y agrega la ruta de instalación de Python, en mi caso sería

![image](https://user-images.githubusercontent.com/36200981/153084396-795b770a-fd59-4904-814a-71484baefda7.png)

De esta manera, podrás utilizar python en cualquier ruta de tu ordenador ya sea utilizando cmd.exe, powershell o cualquier otro


![image](https://user-images.githubusercontent.com/36200981/153084548-b5d45915-63d9-49ab-b089-15fdf03be145.png)


Espero te sea de utilidad, saludos colega,

**Reply by DiegoGomezL03** [reply]
Muchas Gracias @rickygzz, tu explicación fue muy buena. Ahora funciona.  🥇

**Reply by AngelCruzO** [reply]
Cuando utilicen el instalador de la pagina oficial seleccionen la casilla que dice "Add Python 3.10 to PATH". Eso les genera las variables de entorno de manera automática

**Reply by LisandroHdz** [reply]
py -m venv env
a mi se me activo utlizando solo py -m venv env

**Reply by LisandroHdz** [reply]
https://docs.google.com/document/d/1-iMjH-zFETXOu6npVMdgIbjmLFSE8zQKv0Y-b5nULOc/edit?usp=sharing 
te dejo el link de mi ejercicio por si te ayuda, saludos

**Reply by MauricioNovelo** [reply]
prueba con virtualenv env
primero tienes que instalar virtualenv
pip install virtualenv
luego de eso ya puede usar el comando
virtualenv (nombre de tu entorno)

**Reply by MauricioNovelo** [reply]
https://www.youtube.com/watch?v=mQ0IUy_18mI&t=783s&ab_channel=SoloPython cehquen este video, desde el minuto 10 ya les enseña sobre como realizarlo

**Reply by elielestevez** [reply]
@rickygzz muchas gracias hermano me hiciste un parote.

**Reply by AlfredoLabra** [reply]
> Cuando utilicen el instalador de la pagina oficial seleccionen la casilla que dice "Add Python 3.10 to PATH". Eso les genera las variables de entorno de manera automática

Gracias me ayudo mucho tu respuesta, reinstale y coloque la opcion de Añadir el PATH, Saludos!

**Reply by Karen-Rodriguez** [reply]
hola @rickygzz seguí todos los pasos que mencionas y me sigue saliendo este error :(

![image](https://user-images.githubusercontent.com/71851591/153731572-b561eeea-e1de-4e2c-90f8-3bc6c9d8fa7f.png)

**Reply by rickygzz** [reply]
Ahh ya lo lograste!

Para salir de ahi escribe exit() y Enter


En lugar de poner python debe ser python --version

**Reply by Karen-Rodriguez** [reply]
@rickygzz  no puedo pasar al siguiente punto :(

![image](https://user-images.githubusercontent.com/71851591/153731974-74da1950-12d5-4f1d-9343-b25c57e2aaf4.png)

**Reply by rickygzz** [reply]
En esa carpeta pon los siguientes comandos (cada línea con Enter)

cd env
cd Scripts
activate

Y ya te debe aparecer (env) al iniciar la línea

PD si no funciona activate intenta activate.bat

Me dices si funciona (estoy lejos de la computadora)

**Reply by Karen-Rodriguez** [reply]
@rickygzz ya me salió!!! muchas gracias por tu ayuda jaja ya estaba a punto de llorar

**Reply by oscardavnr** [reply]
No le pongas python3, usa py

**Reply by ErickDominguez95** [reply]
> Hola Diego,
> 
> Imagino que descargaste la última versión de Python 3.10.2 (https://www.python.org/downloads/). Al instalarla te dio a elegir el folder de instalación
> 
> ![image](https://user-images.githubusercontent.com/36200981/153082860-8d9c4a53-02a3-4324-b545-ee4f395d97d3.png)
> 
> En mi caso me sugiere la ruta `C:\Users\PC\AppData\Local\Programs\Python\Python310\` (mi nombre de usuario es PC, si tu nombre de usuario para entrar a Windows es "Diego" entonces la ruta sugerida sería `C:\Users\Diego\AppData\Local\Programs\Python\Python310\`). Cualquiera que haya sido la ruta de instalación, comprueba en el explorador de Windows si esa ruta contiene los archivos ejecutables.
> 
> ![image](https://user-images.githubusercontent.com/36200981/153083356-d813b9cf-1eaf-4a11-a539-3eff0ed9c71c.png)
> 
> Una vez confirmada la ruta, copiala con Ctrl + C. En el explorador de Windows, ubica `Mi PC`, da clic derecho, Propiedades, Configuraciones de sistema avanzadas
> 
> ![image](https://user-images.githubusercontent.com/36200981/153083837-cf370ce3-7357-48a7-a254-0d80196f51ee.png)
> 
> Te abre un cuadro de diálogo, en el tab Avanzado, da clic en el botón Variables de Entorno
> 
> ![image](https://user-images.githubusercontent.com/36200981/153084017-d8fe9d69-6302-4e01-b391-cef62278b18c.png)
> 
> En el nuevo cuadro de diálogo, selecciona la variable Path y editar
> 
> ![image](https://user-images.githubusercontent.com/36200981/153084281-868f8d8c-ae2d-4a7b-bac8-6725bf1adce5.png)
> 
> Y agrega la ruta de instalación de Python, en mi caso sería
> 
> ![image](https://user-images.githubusercontent.com/36200981/153084396-795b770a-fd59-4904-814a-71484baefda7.png)
> 
> De esta manera, podrás utilizar python en cualquier ruta de tu ordenador ya sea utilizando cmd.exe, powershell o cualquier otro
> 
> ![image](https://user-images.githubusercontent.com/36200981/153084548-b5d45915-63d9-49ab-b089-15fdf03be145.png)
> 
> Espero te sea de utilidad, saludos colega,

muchas gracias esta info me sirvio mucho

**Comment by FernandaOchoa** [comment]
Hola 🖐️ @GerardoL23 

Ya tiene solución la duda que planteaste.

**Reply by GerardoL23** [reply]
Gracias, acabo de leerla 👯

**Comment by AndreaGerman** [comment]
Hola! me podrían ayudar por favor? :( soy nueva en esto y en este caso estoy utilizando macOS big sur, me encuentro en el siguiente paso pero nose si lo estoy haciendo bien
<img width="1218" alt="Captura de Pantalla 2022-02-08 a la(s) 21 35 06" src="https://user-images.githubusercontent.com/99223392/153123250-06605afe-b05d-4016-837a-2fda4afb7609.png">

**Reply by ArielCanedo** [reply]
Hola, ya instalaste python? puedes probar este código "python -m venv (nombre de la maquina) y luego este para mac source (nombre de la maquina)/bin/activate

Te dejo este link: https://docs.python.org/es/3/tutorial/venv.html
Pero debería funcionar para mac

**Reply by robertomoo** [reply]
Utiliza otra ventana de terminal tiene que salir primero tu usuario, o tal vez no tengas el Python 3, todos los mac vienen con la versión 2.7 instalada

**Reply by lordfriky** [reply]
Si no tienes Python instalado tienes 2 formas de instalarlo:

## Por medio del paquete oficial
Ve a la [web oficial de Python](https://www.python.org/) y **descarga** la última versión disponible (actualmente 3.10.2).

![web_python](https://i.imgur.com/8Y84JHB.png)

Una vez descargado tendrás un archivo con extensión `.pkg`, **ábrelo** e iniciará el programa de instalación.

![instalador_python](https://i.imgur.com/kl4j7sx.png)

La instalación es la típica de darle continuar, continuar, continuar hasta que termine. Una vez terminado podrás comprobar que tienes Python instalado con estos 2 nuevos programas en el Launchpad.

![programas](https://i.imgur.com/G1lW6Ug.png)

De igual manera puedes comprobarlo usando el comando para revisar la versión en la terminal: `python3 --version`. (Nota: Yo tengo una versión un poco vieja instalada, pero en tu terminal se verá la última.)

![python_version](https://i.imgur.com/fpo7Vjf.png)

## Por medio del gestor de paquetes `brew`
Esta es mi preferida ya que por medio de `brew` puedes instalar fácilmente muchos más paquetes que puedas requerir para desarrollar programas y tenerlos todos actualizados de una manera **simple**. Para instalarlo, como dicen las instrucciones en su [web oficial](https://brew.sh/index_es), sólo hace falta **introducir** este comando en la terminal:

```zsh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Nota:** Una vez iniciada la instalación te pedirá tu contraseña para continuar. Si nunca has puesto tu contraseña en la terminal sólo te comento que **no se verá nada** cuando la escribas por razones de seguridad, así que no te preocupes por eso.

![brew_instalacion](https://i.imgur.com/g7jtXhn.png)

Una vez con `brew` instalado sólo tienes que usar el comando `brew install *nombre_del_paquete*` para instalar **cualquier cosa**. En el caso de Python no estoy seguro cuál sea el paquete por defecto actualmente (ya que hace poco era la versión 3.9 siendo que la 3.10 ya había sido lanzada), así que te **recomiendo** usar este comando para instalarlo:

```zsh
brew install python@3.10
```

Una vez terminado puedes comprobar que Python está instalado **de la misma forma** como si lo hubieras instalado con su paquete oficial.

**Finalmente**, si quieres actualizar todos los paquetes que tengas instalados con `brew` *de golpe* puedes usar el comando `brew upgrade`

![brew_upgrade](https://i.imgur.com/gNYb9JL.png)

**Comment by IrmaDiazA** [comment]
Buen día, yo también soy nueva en esto, estoy leyendo la documentación del módulo 2 y honestamente no entiendo qué tengo que hacer, ni cómo debo de hacerlo. Ya no se utiliza visual studio code? estoy perdida.

**Reply by alfredo-inzunza** [reply]
Hola. Te recomiendo ver la [grabación del Live](https://teams.microsoft.com/l/channel/19%3A4SY7l6F1oATeZRTW_mEV7y0PHO9MXkf7aHaJh4PRZXI1%40thread.tacv2/tab%3A%3A558529f6-9c29-4d46-85ff-05c5048a1e98?groupId=52619dee-b1e1-4b4c-9bd4-ea314cca100c&tenantId=4ae54b05-b77e-4224-aef1-8661422e0816) que realizó Fer el Lunes. Yo también voy un poco atrasado pero alli mismo indica que para la Kata 2, las evidencias se generan con capturas de pantalla, ya que en las mismas instrucciones nos piden utilizar la terminal. En general se nos pide crear un entorno virtual, Fer lo explica como si fuera una burbuja, con un entorno seguro para cuando desarrollamos un programa, de esta manera podemos instalar paquetes en este entorno sin afectar otros entornos. 

Tip. Son 3 horas de grabación del Live del día Lunes, yo lo miré en velocidad x2.0, ya cuando llegó a un punto en donde requería prestar más atención, reduje la velocidad de reproducción.

**Comment by DavidLuceroSigcho** [comment]
Buenas Noches a todos, alguien me puede ayudar. 
Tengo problema para activar el entorno virtual, ya hice instrucciones de alias, también de path en resumen hice los otros colegas y no me deja activarlo no se porque ya intente varias formas y nada. Que me ayude como puedo solucionar el problema, se lo agradecería mucho.

![ProblemaEnv](https://user-images.githubusercontent.com/14005639/153353190-f65c2a09-3efc-49e4-97ff-60db5c9bd4ab.png)

**Reply by Angel-Guerra** [reply]
![153077221-8420d46c-2ca2-42ff-9061-db41cd212202.png](https://user-images.githubusercontent.com/99145981/153354715-6a97db01-c2fe-4b01-a7b8-255851b84bf7.png)

Literal lo puse así y jalo

**Reply by Angel-Guerra** [reply]
Si te crea la carpeta env?

**Reply by DavidLuceroSigcho** [reply]
Si amigo e incluso esta todo el Script en la caperta env dentro de archivo.

![image](https://user-images.githubusercontent.com/14005639/153355180-965618ab-ae28-4232-b551-25c90fa9940a.png)

**Reply by Angel-Guerra** [reply]
Ok, ya entendí, te sale como Ami, no estás en cmd vdd?. Usa la terminal cmd por qué en el cmd no se una el ls

**Reply by Angel-Guerra** [reply]
Yo le cambié y ya jalo

**Reply by DavidLuceroSigcho** [reply]
Wow tambien me funciono 😳 pero porque funciona con cmd y no powershell porque sera?. Gracias amigo por tu ayudar.
![image](https://user-images.githubusercontent.com/14005639/153355912-9bcbe28d-694c-4b3d-81e7-412301b14565.png)

**Reply by Angel-Guerra** [reply]
Lo dijeron hoy en el metting cambio de power Shell a cmd. Desconozco la razón pero así le hicieron.

**Reply by DavidLuceroSigcho** [reply]
Un misterio de la vida jeje.

A ti te salio este error cuando escribiste este comando .> pip freeze <. que debo hacer? aun no me acostumbo a cmd no se como borrar jeje.

![image](https://user-images.githubusercontent.com/14005639/153356415-73bcfa88-2d29-41a6-b2e7-949bccb27bc0.png)

**Reply by Angel-Guerra** [reply]
La vdd no me acuerdo. Esque apague la lap hace rato ando desde el cel. Igual reviso en la mañana por qué ya se durmieron todos aquí y no los quiero despertar.

**Reply by DavidLuceroSigcho** [reply]
Lo imagino amigo e igual yo ando en oscuridad como murciélago para no despertar a nadie. 
Porfa como ambos tuvimos el problema es mas fácil porque cada uno de suceder errores diferentes, si no es mucha molesta amigo te lo agradecería mucho 🥺

**Reply by Angel-Guerra** [reply]
Hola, disculpa andaba trabajando, pudiste resolverlo?

**Comment by JesusPantoja1998** [comment]
Una pregunta ¿Alguien sabe porque me aparece acceso denegado cuando quiero instalar el paquete?

![image](https://user-images.githubusercontent.com/99104366/153418824-b46f20d1-8567-4321-b8ac-7b72a31681bf.png)

**Reply by JosueCT** [reply]
> Una pregunta ¿Alguien sabe porque me aparece acceso denegado cuando quiero instalar el paquete?
> 
> ![image](https://user-images.githubusercontent.com/99104366/153418824-b46f20d1-8567-4321-b8ac-7b72a31681bf.png)

Espero que ya se haya resuelto, pero si no... intenta ejecutar CMD como Administrador(Inicio - Busca: CMD - Clic derecho: Ejecutar como administrador), eso debería resolverlo. 😀

**Comment by SaulHernandezM** [comment]
Ayuda e intentado todas las soluciones que dieron y con ninguna me funciono.
![Screenshot 2022-02-12 155133](https://user-images.githubusercontent.com/99091932/153729630-64e85f58-73fd-4e93-9b17-a1d558d5166d.png)

**Reply by alfredo-inzunza** [reply]
Amigo. Tal parece que no tienes Python instalado. Verifica con `python --version` o `python3 --version`, te debe devolver la versión que tienes instalada.  Si no lo tienes instalado, ingresa a la Microsoft Store e instala Python desde alli.

Saludos

**Reply by Karen-Rodriguez** [reply]
hola @alfredo-inzunza  yo tengo el mismo problema y sí tengo isntalado Phyton :( ya instalé y desinstalé pero me sigue saliendo lo mismo 

![image](https://user-images.githubusercontent.com/71851591/153731528-391be8dc-9bc3-48ca-b4ba-cb9ae325b064.png)

**Reply by enrique-mendoza** [reply]
Hola.

En mi caso yo pude realizarlo escribiendo `python` en lugar de `python3` antes de escribir `-m venv env`.

![image](https://user-images.githubusercontent.com/33888244/153731711-de8c98a9-9cba-4927-b2f2-512e77f12d9b.png)

Aquí muestro como hago lo mismo para enlistar los paquetes que tengo instalados en mi entorno.

**Reply by SaulHernandezM** [reply]
![Screenshot 2022-02-12 200827](https://user-images.githubusercontent.com/99091932/153735271-8bda1929-ff57-4e4e-89d8-170f94ab58a5.png)
Ya he intentado todo lo han dicho pero no me funciona. Y si tengo el python instalado, ya no se que hacer.

**Reply by JosueCT** [reply]
> ![Screenshot 2022-02-12 200827](https://user-images.githubusercontent.com/99091932/153735271-8bda1929-ff57-4e4e-89d8-170f94ab58a5.png) Ya he intentado todo lo han dicho pero no me funciona. Y si tengo el python instalado, ya no se que hacer.

Tuve un problema parecido, aunque descargue(https://www.python.org/downloads/) e instale Python, de hecho @SaulHernandezM  tengo la misma versión que tu, lo que hice fue ir a la tienda de Microsoft Store, buscar: **python** e instalar la versión: Python 3.10 

![image](https://user-images.githubusercontent.com/17696440/153742806-49fd5e50-9f2e-44d5-a0b3-a2a25a3074a6.png)

Después volví a intentar con el comando: **python3 -m venv env** , si no funciona, intentalo sin el numero: **python -m venv env**
